src/aeonlib/models.py

```diff
@@ -40,66 +40,134 @@ class SiderealTarget(BaseModel):
-    diff_azimuth_acceleration: float | None = None
-    """Differential azimuth acceleration (arcsec/s^2)"""  # Satellite Only
-    meanlong: Angle | None = None
-    """Mean longitude (angle in degrees)"""  # Major Planet Only
-    longofperih: Angle | None = None
-    """Longitude of perihelion (angle in degrees)"""  # Major Planet Only
     extra_params: dict[Any, Any] = {}
 
 
-class Window(BaseModel):
-    """A general time window"""
+class AsaMajorPlanetTarget(_NonSiderealTarget):
+    scheme: Literal["ASA_MAJOR_PLANET"] = "ASA_MAJOR_PLANET"
+    longofperih: Angle
+    """Longitude of perihelion (angle in degrees)"""
+    meandist: Annotated[float, NonNegativeFloat]
+    """Semi-major axis (AU)"""
+    meanlong: Angle
+    """Mean longitude (angle in degrees)"""
+    dailymot: float
+    """Daily motion (angle in degrees)"""
+
+
+class AsaMinorPlanetTarget(_NonSiderealTarget):
+    scheme: Literal["ASA_MINOR_PLANET"] = "ASA_MINOR_PLANET"
+    argofperih: Angle
+    """Argument of perihelion (angle in degrees)"""
+    meandist: Annotated[float, NonNegativeFloat]
+    """Semi-major axis (AU)"""
+    meananom: Angle
+    """Mean anomaly (angle in degrees)"""
+
+
+class AsaCometTarget(_NonSiderealTarget):
+    scheme: Literal["ASA_COMET"] = "ASA_COMET"
+    argofperih: Angle
+    """Argument of perihelion (angle in degrees)"""
+    perihdist: Annotated[float, NonNegativeFloat]
+    """Perihelion distance (AU)"""
+    epochofperih: TimeMJD
+    """Epoch of perihelion (MJD)"""
 
+
+class JplMajorPlanetTarget(_NonSiderealTarget):
+    scheme: Literal["JPL_MAJOR_PLANET"] = "JPL_MAJOR_PLANET"
+    argofperih: Angle
+    """Argument of perihelion (angle in degrees)"""
+    meandist: Annotated[float, NonNegativeFloat]
+    """Semi-major axis (AU)"""
+    meananom: Angle
+    """Mean anomaly (angle in degrees)"""
+    dailymot: float
+    """Daily motion (angle in degrees)"""
+
+
+class JplMinorPlanetTarget(_NonSiderealTarget):
+    scheme: Literal["JPL_MINOR_PLANET"] = "JPL_MINOR_PLANET"
+    argofperih: Angle
+    """Argument of perihelion (angle in degrees)"""
+    perihdist: Annotated[float, NonNegativeFloat]
+    """Perihelion distance (AU)"""
+    epochofperih: TimeMJD
+    """Epoch of perihelion (MJD)"""
+
+
+class MpcMinorPlanetTarget(_NonSiderealTarget):
+    scheme: Literal["MPC_MINOR_PLANET"] = "MPC_MINOR_PLANET"
+    argofperih: Angle
+    """Argument of perihelion (angle in degrees)"""
+    meandist: Annotated[float, NonNegativeFloat]
+    """Semi-major axis (AU)"""
+    meananom: Angle
+    """Mean anomaly (angle in degrees)"""
+
+
+class MpcCometTarget(_NonSiderealTarget):
+    scheme: Literal["MPC_COMET"] = "MPC_COMET"
+    argofperih: Angle
+    """Argument of perihelion (angle in degrees)"""
+    perihdist: Annotated[float, NonNegativeFloat]
+    """Perihelion distance (AU)"""
+    epochofperih: TimeMJD
+    """Epoch of perihelion (MJD)"""
+
+
+class SatelliteTarget(BaseModel):
```

Comment:
This might be worth a name change.
I feel like we need to emphasize the "artificial" part of Artificial satellite.

There is an additional type of non-sidereal object not accounted for here (because it's hard) that's "natural satellite" with a whole other set of elements, to let you know where another planet's moon will be in the sky at any given time. This is actually quite important when performing natural satellite spectroscopy because you need to align the slit to catch the moon and miss the planet.

We should not worry about adding this as a type for the moment, but it might be nice to leave the namespace open for the enterprising Uranian satellite scientists of the future.

Reply:
What's a good name for them? GeocentricSatellite? I'm assuming non-earth satellites can be modeled as asteroids or something?

Reply:
I think that's great. I doubt we'll often be in a position to look at non-earth artificial satellites, but GeocentricSatellite is more precise. (and who knows there could be temporary natural satellites that need to be described that way. 

Another option would be "TripletSatelliteTarget" or something to describe the schema for locating them (the position/velocity/acceleration values are known as a triplet) in the same way that MPC and JPL use different expected schema.